### DISCUSSION TITLE
building U-Boot

### DISCUSSION THREAD

**Discussion by rockoguido** [polarfire-soc/polarfire-soc-documentation#340] [HSS and U-Boot]
Hi .. looks like this has not come up yet , did not see it in the history here.

Helping a customer who potentially wants to port out U-Boot for the PFSoC to their custom platform.   Can't really tell if any porting work needs to be done but just wanted to get comfortable with the process.   
So, went to github.com/polarfire-soc/u-boot and cloned that repo..   
Can configure for the Icicle Kit via make make microchip_mpfs_icicle_defconfig without an issue but of course the actually build process via make fails at the first compilation which makes sense since the tools have not been setup.
I have some cross tools on my machine and updated my path to them but I suspect there is more to do..
I can look at the Makefiles to sort it out but question is there any process where this is documented to all the user to build U-Boot?
Thx!
Dave

**Comment by padmaraob** [comment]
Documented here [https://github.com/u-boot/u-boot/blob/master/doc/board/microchip/mpfs_icicle.rst](https://github.com/u-boot/u-boot/blob/master/doc/board/microchip/mpfs_icicle.rst)

**Reply by rockoguido** [reply]
belated thanks!   works great.   I used the cross tools that I built via buildroot, was able to build without an issue
